### ISSUE TITLE
SC problematic for Closed functionality: 1.4.5 Images of Text

### ISSUE THREAD

**Issue by pday1** [w3c/wcag2ict#276]
From closed functionality questionnaire:
https://www.w3.org/2002/09/wbs/55145/wcag2ict-sc-problematic-for-closed/results#xq25

Mitchell Evan	|	1.4.5 Images of Text: Revise as follows:
Requires text for interoperability with assistive technology and high-quality resizing. Where other mechanisms achieve equivalent results, this criterion does not apply.

**Comment by pday1** [comment]
SC, and Applying SC text included for reference

> 1.4.5 Images of Text
> If the technologies being used can achieve the visual presentation, text is used to convey information rather than images of text except for the following:
> 
> Customizable
> The image of text can be visually customized to the user's requirements;
> 
> Essential
> A particular presentation of text is essential to the information being conveyed.
> 
> Logotypes (text that is part of a logo or brand name) are considered essential.

> Applying SC 1.4.5 Images of Text to Non-Web Documents and Software
> 
> This applies directly as written, and as described in [Intent from Understanding Success Criterion 1.4.5](https://www.w3.org/WAI/WCAG22/Understanding/images-of-text#intent) (also provided below).
> 
> NOTE
> See also the discussion on [Closed Functionality](https://w3c.github.io/wcag2ict/#closed-functionality).

**Comment by pday1** [comment]
@mitchellevan I've taken your proposed text from the survey results, and added an example as I thought that it might be helpful in the context of closed systems. See what you think.

**Comment by mraccess77** [comment]
The proposed wording addresses resize of text being clear but not customization such as following the platforms contrast settings - e.g. high contrast or dark mode - this is akin to text being used which can be controlled by CSS in HTML.

In addition, I'm not sure why audio description/audio access is included with this.

It seems like this is potentially creating an exception to allow for images of text in closed functionality software - but why is that exception needed?  Wouldn't the existing exception be sufficient?  Use of text rather than images of text in closed functionality would still be useful to users who need that.

**Comment by maryjom** [comment]
@mraccess77 The Task Force is proposing an update to the existing 2013 guidance for the SC problematic for closed functionality which originally said:
>1.4.5 Images of Text—because there is no need to impose a requirement on all closed functionality that text displayed on the screen actually be represented internally as text (as defined by WCAG 2.0), given that there is no interoperability with assistive technology;

Not all closed functionality products have a fully functioning operating system. There isn't a screen reader, and in those cases it is likely there is no functionality to adjust anything about the display or contrast of any contents displayed, including an image. Instead, these products have requirements (set by 508 and the EN 301 549) to self-voice which would provide aurally any information the image is conveying as an accessible alternative. IMO, such a case doesn't meet either of the stated exceptions for the SC. The technology can display text so the SC clause "If the technologies being used can achieve the visual presentation" would seem to never apply - even to Web content.

**Comment by mraccess77** [comment]
I think we are making assumptions that closed functionality can't or won't be able to allow users to use dark mode or adjust settings.  Rather than giving a blanket exception for closed functionality - I'd rather say if these things aren't available then there is an exception.  We already have an exception for technology allowance at AA - so I'd rather only introduce that if it's truly not possible to do something.  Whether there is a screen reader or not isn't so relevant to the criteria in my opinion as this is focused on users with low vision and cognitive disabilities to visually control/enlarge the text.

**Comment by GreggVan** [comment]
Good point  Jonathan.   And it would extend beyond  dark mode.   there are lots of built in accessibility features.     for closed products though -- if we say that you can use the built-in or you need to supply your own... which is what we normally say -- then it would mean that closed products would need to do more --or let what is there flow through more.     this is a really tough area for some closed products -- even though kind of clear and obvious for others.   Good thing we don't have to solve closed products - just note which SC won't work as designed for closed products.

**Comment by pday1** [comment]
For ease of review, here is content pulled from the latest Editor's draft to show the original "Applying SC to Non-Web..." alongside the proposed new content for "SC problematic for Closed...".

The latest editor's draft can be found at https://w3c.github.io/wcag2ict/
(Note this does not contain content that is in pull requests that are open and not yet approved)

**Content from Applying SC**





**Latest proposed content for SC problematic for Closed functionality section**

**Comment by mitchellevan** [comment]
My survey response, which started this issue, was from July 2023. Since then, I've come to understand we are not expected to draw a bright line between pass and fail for each SC in [Success Criteria Problematic for Closed Functionality](https://www.w3.org/TR/wcag2ict-22/#success-criteria-problematic-for-closed-functionality).

So I'll offer the following option:

> 1.4.5 Images of Text—For content that meets the "can achieve" condition and does not meet the "customizable" or "essential" exception, this Success Criterion requires text in a programmatically determinable form.

This option is more consistent with our other "problematic" SCs that rely on "programmatically determinable", such as:

> 1.1.1 Non-text Content—requires text or a text alternative

> 1.3.2 Meaningful Sequence—requires information in a programmatically determinable form

**Comment by mraccess77** [comment]
I'm not sure just being available programmatically addresses the intent and wording of the criterion. That - I believe the intent was that the display of it could be adjusted.  If the intent was to just make it available then there really would not have been a need for SC 1.4.5 at all.   it seems like the existing exceptions are sufficient to allow passes for many closed functionality systems where it is not possible.

**Comment by mitchellevan** [comment]
Thanks @mraccess77! I will make sure the Task Force examines my proposal for accuracy and clarity in context of "problematic for closed." I will also make sure we discuss dependency on underlying platforms for text customizations.

> I'm not sure just being available programmatically addresses the intent and wording of the criterion. That - I believe the intent was that the display of it could be adjusted. If the intent was to just make it available then there really would not have been a need for SC 1.4.5 at all.

My proposal today is consistent with normative SC 1.4.5. This SC requires "text", defined as a "sequence of characters that can be programmatically determined...". While the intent of the SC is adaptability of visible text, the purpose of the current issue is not to reiterate the non-problematic aspects of the SC, but only to highlight what can be problematic for closed functionality software.

> it seems like the existing exceptions are sufficient to allow passes for many closed functionality systems where it is not possible.

Is there an example that differs substantially from web? I can't think of any.

* I've never seen the "customizable" exception actually met, but it seems just as feasible in open and closed software as it would be in web content.
* The "essential" exception depends on the purpose of the content, not the technology's capability, so this exception applies equally to open and closed software and to web content.
* I can't say how often the "can achieve" condition occurs on various open and closed platforms. Non-web platforms could be more or less capable than web; for example, it could be easier to license fonts. Software designers and developers might use raster and vector images more often, or they might use them less. Frequency notwithstanding, the "can achieve" condition is in principle the same for any platform.

Bottom line, the only thing "problematic for closed" in SC 1.4.5 seems to be the requirement that text be programmatic.

**Comment by mraccess77** [comment]
@mitchellevan - thank you for considering this -  after thinking about this more - while text is defined as programmatic sequence of characters in a human language and I am ok with us clearly communicating it's programmatically determinable - the key point is that that "text" be used instead of the image of text and not as an alternative means of access to that image of text.  If the SC only only required programmatic access then alt text would always meet the requirement. But the crux of the SC is "rather than".

So perhaps something like "when an exception does not apply this success criterion requires text be programmatically determinable rather than relying on the image of text".

I believe the can achieve clauses could be used to cover situations in closed functionality systems where presentation with system text is problematic.

**Comment by maryjom** [comment]
Added Mitch's proposals to a [survey](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-SC-problematic-remaining-sc/) that will open on 8 March. On that day the TF will work on Name, Role Value, Info and relationships SC guidance for Closed Functionality which will also be in this same survey.

**Comment by maryjom** [comment]
TF reached consensus on 11 April. SC Problematic for Closed: 1.4.5 Images of Text - [Incorporate Option 12 as shown in the meeting minutes.](https://www.w3.org/2024/04/11-wcag2ict-minutes#r03)

See PR #337, which has been merged.
